lib/spack/docs/configuration.rst

```diff
@@ -221,3 +269,145 @@ The syntax for the ``provider`` section differs slightly from other
 concretization rules.  A provider lists a value that packages may
 ``depend_on`` (e.g, mpi) and a list of rules for fulfilling that
 dependency.
+
+
+.. _configuration-scopes:
+
+-------------------------
+Configuration Scopes
+-------------------------
+
+Spack pulls configuration data from files in several directories. There
+are three configuration scopes.  From lowest to highest:
+
+1. **defaults**: Stored in ``$(prefix)/etc/spack/defaults/``. These are
+   the "factory" settings. Customizations should be made in the other
+   higher precedence scopes, as defaults may change from version to
+   version.
+
+2. **site**: Stored in ``$(prefix)/etc/spack/``.  Settings here affect
+   only *this instance* of Spack.  They override ``defaults``.  The site
+   scope can can be used for per-project settings (one spack instance per
+   project) or for site-wide settings on a multi-user machine (e.g., for
+   a common spack instance).
+
+3. **user**: Stored in the home directory, ``~/.spack/``. These settings
+   affect all instances of Spack and take the highest precedence.
+
```

Comment:
We also need a **project** scope.  Typical use case... I'm building two software stacks with different version/dependency selections in them.  Or I want to run an integration test with a `packages.yaml` different from my "normal" one.  Suggested ways to find the project scope would be:

1. An env var

2. A command line argument.  (This would have to be available on _all_ commands, ideally parsed and set up before the command is run.  So then one could `alias myproject-spack 'spack --config ~/myproject/dotspack'`

3. Search up the source tree for a `dotspack` directory.

I would lean toward (2) as probably the best way to do this.  As long as the Bash alias works, users  can easily set up as many per-project Spack-like commands as they need.